polkadot/runtime/westend/src/lib.rs

```diff
@@ -2811,6 +2819,114 @@ mod remote_tests {
 			.unwrap();
 		ext.execute_with(|| Runtime::on_runtime_upgrade(UpgradeCheckSelect::PreAndPost));
 	}
+
+	#[tokio::test]
+	async fn delegate_stake_migration() {
+		// Intended to be run only manually.
+		if var("RUN_MIGRATION_TESTS").is_err() {
+			return;
+		}
+		use frame_support::assert_ok;
+		sp_tracing::try_init_simple();
+
+		let transport: Transport = var("WS").unwrap_or("ws://127.0.0.1:9900".to_string()).into();
+		let maybe_state_snapshot: Option<SnapshotConfig> = var("SNAP").map(|s| s.into()).ok();
+		let mut ext = Builder::<Block>::default()
+			.mode(if let Some(state_snapshot) = maybe_state_snapshot {
+				Mode::OfflineOrElseOnline(
+					OfflineConfig { state_snapshot: state_snapshot.clone() },
+					OnlineConfig {
+						transport,
+						state_snapshot: Some(state_snapshot),
+						pallets: vec![
+							"staking".into(),
+							"system".into(),
+							"balances".into(),
+							"nomination-pools".into(),
+							"delegated-staking".into(),
+						],
+						..Default::default()
+					},
+				)
+			} else {
+				Mode::Online(OnlineConfig { transport, ..Default::default() })
+			})
+			.build()
+			.await
+			.unwrap();
+		ext.execute_with(|| {
+			// create an account with some balance
```

Comment:
If you want to easily reuse this code for other runtimes, should this part onward live in the pallet, generic over `<T: Runtime>`?

Reply:
or maybe you are fine with copy-pasting it.

Reply:
By adding it in the westend runtime, I get a runtime configuration closer to mainnets (kusama/polkadot). For example: The nomination pool pallet test runtime config does not even include pallet-staking.

I have been generally reusing this by just pointing to the polkadot or kusama node with ED the only thing that differs between the runtimes relevant for this test. The most accurate test though would be by copy pasting. Does that make sense?